Response/DatatableQueryBuilder.php

```diff
@@ -263,8 +264,14 @@ private function initColumnArrays()
                     $currentPart = array_shift($parts);
                     $currentAlias = ($previousPart === $this->entityShortName ? '' : $previousPart.'_').$currentPart;
 
-                    if (!array_key_exists($previousAlias.'.'.$currentPart, $this->joins)) {
-                        $this->addJoin($previousAlias.'.'.$currentPart, $currentAlias, $this->accessor->getValue($column, 'joinType'));
+                    $columnTableName = $previousAlias.'.'.$currentPart;
+                    if (!array_key_exists($columnTableName, $this->joins)) {
+                        $this->addJoin($columnTableName, $currentAlias, $this->accessor->getValue($column, 'joinType'));
+                    }
+                    // for the last dql part join using WITH, if expression is given for column
+                    if (count($parts) === 1 && array_key_exists($columnTableName, $this->joins) && null !== $withExpr) {
+                        $with = str_replace($currentPart . '.', $currentAlias . '.', $withExpr);
+                        $this->addJoin($columnTableName, $currentAlias, $this->accessor->getValue($column, 'joinType'), $with);
```

Comment:
I you have a "complex" join like `table1.table2.table3.field`, maybe we want to set a different join condition on each join.
How do you handle it?

Reply:
That's possible by adding a column of each joined table to the column builder and defining the respective join condition.

```
$this->columnBuilder
    ->add('table1.table2.id', Column\Column::class, array(
        'visible' => false,
        'join_conditions' => 'table2.enabled = 1'
    ));
```

Reply:
Did you tried it?
Is the `table2.enabled = 1` not applied on the `table1.table2` join?